code/weapon/weapons.cpp

```diff
@@ -5718,27 +5734,28 @@ void weapon_process_post(object * obj, float frame_time)
 
 
 	// check life left.  Multiplayer client code will go through here as well.  We must be careful in weapon_hit
-	// when killing a missile that spawn child weapons!!!!
+	// when killing a missile that spawns child weapons!!!!
 	if ( wp->lifeleft < 0.0f ) {
-		if ( wip->subtype & WP_MISSILE ) {
-			if(Game_mode & GM_MULTIPLAYER){				
-				if ( !MULTIPLAYER_CLIENT || (MULTIPLAYER_CLIENT && (wp->lifeleft < -2.0f)) || (MULTIPLAYER_CLIENT && (wip->wi_flags[Weapon::Info_Flags::Child]))) {					// don't call this function multiplayer client -- host will send this packet to us
+		if (wip->wi_flags[Weapon::Info_Flags::Detonate_on_expiration]) {
+			if (Game_mode & GM_MULTIPLAYER && wip->subtype & WP_MISSILE) {
+				// don't call this function if multiplayer client -- host will send this packet to us
+				if ( !MULTIPLAYER_CLIENT || (MULTIPLAYER_CLIENT && (wp->lifeleft < -2.0f)) || (MULTIPLAYER_CLIENT && (wip->wi_flags[Weapon::Info_Flags::Child]))) {
 					weapon_detonate(obj);					
 				}
 
 				if (MULTIPLAYER_MASTER) {
 					send_missile_kill_packet(obj);
 				}
-
 			} else {
 				weapon_detonate(obj);
 			}
-			if (wip->is_homing()) {
-				Homing_misses++;
-			}
 		} else {
             obj->flags.set(Object::Object_Flags::Should_be_dead);
 		}
+
+		if (wip->is_homing()) {
+			Homing_misses++;
+		}
```

Comment:
I'm not 100% sure this is intentional.  Previously this was triggered if `( wip->subtype & WP_MISSILE ) {` .  So this should either be moved back to somewhere it has the equivalent behavior, or changed to `if ( wip->subtype & WP_MISSILE  && wip->is_homing) {`

Reply:
It's intentional, since primary weapons can home now.  But upon taking a closer look at that variable, its only use is to provide information for an `nprintf` that is now commented out.  So I'll remove it.

Reply:
All right.  Yeah, any coder that needs that information to balance AI can recreate the statements pretty easily.